### PR TITLE
check-webkit-style: validate enums declared with explicit type

### PR DESCRIPTION
**Pull request by yury-s**
#### 3acf37a5c0dfe9153dae2dacec650ccafe590b17
<pre>
check-webkit-style: validate enums declared with explicit type
<a href="https://bugs.webkit.org/show_bug.cgi?id=240478">https://bugs.webkit.org/show_bug.cgi?id=240478</a>

Reviewed by Devin Rousso.

check-webkit-style now validates case of the enums delcared with explicit
type specifier.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(_EnumState.process_clean_line):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(NoNonVirtualDestructorsTest):

Canonical link: <a href="https://commits.webkit.org/250659@main">https://commits.webkit.org/250659@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@294351">https://svn.webkit.org/repository/webkit/trunk@294351</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/Scripts/webkitpy/style/checkers/cpp.py [hunk 1]
```diff
@@ -1371,7 +1371,7 @@ def process_clean_line(self, line):
             self.in_enum_decl = True
             self.enum_decl_name = matched.group('identifier')
         else:
-            matched = match(expr_enum_start + r'(?P<members>.*)' + expr_enum_end + r'$', line)
+            matched = match(expr_enum_start + r'(?P<members>[^{]*)' + expr_enum_end + r'$', line)
```
Comment (dcrousso) [inline Tools/Scripts/webkitpy/style/checkers/cpp.py]: why is this needed?

Comment (yury-s) [reply Tools/Scripts/webkitpy/style/checkers/cpp.py]: Otherwise it will greedily match the type as part of the first element: ` : uint8_t { Uno`

Comment (yury-s) [reply Tools/Scripts/webkitpy/style/checkers/cpp.py]: I considered making `{`  required in `expr_enum_start` but there are enum declarations in the code that have `{` on the next line.

**Review by dcrousso — APPROVED** [review]
